test/cram/pattern.t

```
@@ -1,20 +1,87 @@
+
+
+
+
+Testing character classes
+
+  $ simple.c.bin --pattern 'misc/[pf]assing' --verbose
+  [\x1b[0;34m----\x1b[0m] Criterion v2.2.1 (esc)
+  [\x1b[0;34m====\x1b[0m] Running \x1b[0;34m2\x1b[0m tests from \x1b[0;33mmisc\x1b[0m: (esc)
+  [\x1b[0;34mRUN \x1b[0m] misc::passing (esc)
+  [\x1b[0;32mPASS\x1b[0m] misc::passing (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::failing: Test is disabled (esc)
+  [\x1b[0;34m====\x1b[0m] \x1b[0;1mSynthesis: Tested: \x1b[0;34m1\x1b[0;1m | Passing: \x1b[0;32m1\x1b[0;1m | Failing: \x1b[0m0\x1b[0;1m | Crashing: \x1b[0m0\x1b[0;1m \x1b[0m (esc)
+
+  $ simple.c.bin --pattern 'misc/[p-s]assing' --verbose
+  [\x1b[0;34m----\x1b[0m] Criterion v2.2.1 (esc)
+  [\x1b[0;34m====\x1b[0m] Running \x1b[0;34m2\x1b[0m tests from \x1b[0;33mmisc\x1b[0m: (esc)
+  [\x1b[0;34mRUN \x1b[0m] misc::passing (esc)
+  [\x1b[0;32mPASS\x1b[0m] misc::passing (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::failing: Test is disabled (esc)
+  [\x1b[0;34m====\x1b[0m] \x1b[0;1mSynthesis: Tested: \x1b[0;34m1\x1b[0;1m | Passing: \x1b[0;32m1\x1b[0;1m | Failing: \x1b[0m0\x1b[0;1m | Crashing: \x1b[0m0\x1b[0;1m \x1b[0m (esc)
+
+  $ simple.c.bin --pattern 'misc/[!f]ailing' --verbose
+  [\x1b[0;34m----\x1b[0m] Criterion v2.2.1 (esc)
+  [\x1b[0;34m====\x1b[0m] Running \x1b[0;34m2\x1b[0m tests from \x1b[0;33mmisc\x1b[0m: (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::passing: Test is disabled (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::failing: Test is disabled (esc)
+  [\x1b[0;34m====\x1b[0m] \x1b[0;1mSynthesis: Tested: \x1b[0;34m0\x1b[0;1m | Passing: \x1b[0;32m0\x1b[0;1m | Failing: \x1b[0m0\x1b[0;1m | Crashing: \x1b[0m0\x1b[0;1m \x1b[0m (esc)
+
+  $ simple.c.bin --pattern 'misc/[p-f]ailing' --verbose
+  [\x1b[0;34m----\x1b[0m] Criterion v2.2.1 (esc)
+  [\x1b[0;34m====\x1b[0m] Running \x1b[0;34m2\x1b[0m tests from \x1b[0;33mmisc\x1b[0m: (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::passing: Test is disabled (esc)
+  [\x1b[0;33mSKIP\x1b[0m] misc::failing: Test is disabled (esc)
```

(Plus it would confuse people for the pattern to silently fail where the error is obviously an human error)